src/compile/data/optimize.ts

```diff
@@ -41,11 +41,12 @@ function cloneSubtree(facet: FacetNode) {
  * Move facet nodes down to the next fork or output node. Also pull the main output with the facet node.
  * After moving down the facet node, make a copy of the subtree and make it a child of the main output.
  */
-function moveFacetDown(node: DataFlowNode) {
+function moveFacetDown(node: DataFlowNode): boolean {
+  let flag = false;
```

Comment:
Might want to rename `flag` to something like `mutatedFlag` to make it clear what we're tracking

Reply:
Done.